client/blocks/comment-button/index.jsx

```diff
@@ -20,51 +19,57 @@ import { getPostTotalCommentsCount } from 'calypso/state/comments/selectors';
 import './style.scss';
 
 function CommentButton( props ) {
-	const { commentCount, href, onClick, showLabel, tagName, target } = props;
+	const { commentCount, href, onClick, size, target } = props;
 	const translate = useTranslate();
 
-	return React.createElement(
-		tagName,
-		omitBy(
-			{
-				className: 'comment-button',
-				href: 'a' === tagName ? href : null,
-				onClick,
-				target: 'a' === tagName ? target : null,
-			},
-			( prop ) => prop === null
-		),
-		<Gridicon icon="comment" size={ props.size } className="comment-button__icon" />,
-		<span className="comment-button__label">
-			{ commentCount > 0 && <span className="comment-button__label-count">{ commentCount }</span> }
-			{ showLabel && commentCount > 0 && (
-				<span className="comment-button__label-status">
-					{ translate( 'Comment', 'Comments', {
-						context: 'noun',
-						count: commentCount,
-					} ) }
-				</span>
-			) }
-		</span>
+	return (
+		<a className="comment-button" href={ href } onClick={ onClick } tabIndex="0" target={ target }>
```

Comment:
A few things:
1. `tabIndex="0"` shouldn't be necessary if it's already a focusable element.
2. If we're locking this into being an `<a>`, then the naming of `comment-button` is misleading since it's not a button. 
3. Why are we locking this into being an `<a>`? Do we have any areas where we're using this comment button to reveal a comment section (not linked)?